### PR TITLE
new module fargene

### PR DESCRIPTION
**Pull request by louperelo**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x ] Remove all TODO statements.
- [ x] Emit the `versions.yml` file.
- [x ] Follow the naming conventions.
- [x ] Follow the parameters requirements.
- [x ] Follow the input/output options guidelines.
- [x ] Add a resource `label`
- [x ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/fargene/main.nf [hunk 1]
```diff
@@ -0,0 +1,64 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName; getProcessName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+def VERSION = '0.1'
```
Comment (lescai) [inline modules/fargene/main.nf:7]: is there any way to print the version natively?

File: modules/fargene/main.nf [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    tuple val(meta), path("${prefix}/hmmsearchresults/*.out"),                                  optional: true, emit: hmm
+    tuple val(meta), path("${prefix}/predictedGenes/predicted-orfs.fasta"),                     optional: true, emit: orfs
+    tuple val(meta), path("${prefix}/predictedGenes/predicted-orfs-amino.fasta"),               optional: true, emit: orfs_amino
+    tuple val(meta), path("${prefix}/predictedGenes/retrieved-contigs.fasta"),                  optional: true, emit: contigs
+    tuple val(meta), path("${prefix}/predictedGenes/retrieved-contigs-peptides.fasta"),         optional: true, emit: contigs_pept
+    tuple val(meta), path("${prefix}/predictedGenes/*filtered.fasta"),                          optional: true, emit: filtered
+    tuple val(meta), path("${prefix}/predictedGenes/*filtered-peptides.fasta"),                 optional: true, emit: filtered_pept
+    tuple val(meta), path("${prefix}/retrievedFragments/all_retrieved_*.fastq"),                optional: true, emit: fragments
+    tuple val(meta), path("${prefix}/retrievedFragments/retrievedFragments/trimmedReads/*.fasta"), optional: true, emit: trimmed
```
Comment (lescai) [inline modules/fargene/main.nf:38]: the indent should be fixed here, I'd commit the suggestion myself but the browser editor doesn't display the suggestion clearly

File: modules/fargene/main.nf [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    tuple val(meta), path("${prefix}/spades_assembly/*"),                                       optional: true, emit: spades
+    tuple val(meta), path("${prefix}/tmpdir/*.fasta"),                                          optional: true, emit: metagenome
+    tuple val(meta), path("${prefix}/tmpdir/*.out"),                                            optional: true, emit: tmp
+
+    path "versions.yml"          , emit: versions
```
Comment (lescai) [inline modules/fargene/main.nf:43]: and indent here as well

File: modules/fargene/main.nf [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    fargene \\
+        $options.args \\
+        -p $task.cpus \\
+        -i unziped.fa \\
+        --hmm-model $hmm_model\\
+        -o $prefix \\
+
+    cat <<-END_VERSIONS > versions.yml
+    ${getProcessName(task.process)}:
+        ${getSoftwareName(task.process)}: \$(echo $VERSION)
```
Comment (lescai) [inline modules/fargene/main.nf:61]: see comment above, whether is possible to print native version with fargene commands

**Review by lescai — CHANGES_REQUESTED** [review]
few indent changes, and a double check on version output

**Review by drpatelh — APPROVED** [review]
LGTM! 🚀

**Review by lescai — APPROVED** [review]

File: modules/fargene/main.nf [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    conda (params.enable_conda ? "bioconda::fargene=0.1" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/fargene:0.1--py27h21c881e_4"
+    } else {
+        container "quay.io/biocontainers/fargene:0.1--py27h21c881e_4"
+    }
+
+    input:
+    tuple val(meta), path(input) //input may be fasta (for genomes or longer contigs) or paired-end fastq (for metagenome), the latter in addition with --meta flag
+    val(hmm_model)
```
Comment (drpatelh) [inline modules/fargene/main.nf:25]:
```suggestion
    // input may be fasta (for genomes or longer contigs) or paired-end fastq (for metagenome), the latter in addition with --meta flag
    tuple val(meta), path(input) 
    val hmm_model
```

**Pull request merged**